### PR TITLE
Add sample node output symbols pruning rule

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PruneSampleChildrenColumns.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import io.prestosql.sql.planner.plan.PlanNode;
+import io.prestosql.sql.planner.plan.SampleNode;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static io.prestosql.sql.planner.iterative.rule.Util.restrictChildOutputs;
+import static io.prestosql.sql.planner.plan.Patterns.sample;
+
+public class PruneSampleChildrenColumns
```
Comment (martint) [inline presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PruneSampleChildrenColumns.java]: PruneSampleColumns

File: presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestPruneSampleChildrenColumns.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    @Test
+    public void testAllOutputsReferenced()
+    {
+        tester().assertThat(new PruneSampleChildrenColumns())
+                .on(p -> buildProjectedSample(p, alwaysTrue()))
+                .doesNotFire();
+    }
+
+    private ProjectNode buildProjectedSample(PlanBuilder planBuilder, Predicate<Symbol> projectionFilter)
```
Comment (martint) [inline presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestPruneSampleChildrenColumns.java]: This level of abstraction does not make the code easier to read and reason about. Inline in both places where it's used and create the Assignments directly as they are expected by the test.

File: presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestPruneSampleColumns.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        tester().assertThat(new PruneSampleColumns())
+                .on(p -> p.project(
+                        Assignments.identity(p.symbol("a"), p.symbol("b")),
+                        p.sample(
+                                0.5,
+                                SYSTEM,
+                                p.values(p.symbol("a"), p.symbol("b"))))).doesNotFire();
+    }
+
+    private ProjectNode buildProjectedSample(PlanBuilder planBuilder, Predicate<Symbol> projectionFilter)
```
Comment (martint) [inline presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestPruneSampleColumns.java:69]: This method is not used anywhere

Comment (Praveen2112) [reply presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestPruneSampleColumns.java]: Thanks for the feedback !! Applied the changes

**Pull request merged**